src/components/InteractiveMap.tsx

```diff
@@ -1,66 +1,189 @@
-    fillColor: string;
-    fillOpacity: number;
-    color: string;
-    weight: number;
-}
+import { RefObject, useCallback, useMemo, useRef, useState } from "react";
+import DataLoader, { groupBy, GroupBy, IDataEntry } from "../utils/DataLoader";
+import { Colors, getTint } from "../utils/Colors";
+import { LayerTypes } from "../utils/Helper";
 
 const InteractiveMap: () => JSX.Element = () => {
-    const defaultCountryStyle: ICountryStyle = {
-        fillColor: "red",
-        fillOpacity: 0.5,
-        color: "white",
-        weight: 1,
-    };
-    const hoverCountryStyle: ICountryStyle = {
-        fillColor: "red",
-        fillOpacity: 1,
-        color: "white",
-        weight: 3,
-    };
-
-    const onEachFeature = (feature: Feature, layer: Layer): void => {
-        const name = feature.properties?.NAME_2;
-        layer.bindPopup(name);
-        layer.on({
-            mouseover: (event: LeafletEvent) => {
-                event.target.setStyle(hoverCountryStyle);
-                layer.openPopup();
-            },
-            mouseout: (event: LeafletEvent) => {
-                event.target.setStyle({
-                    fillColor: "red",
-                    fillOpacity: 0.5,
-                    color: "white",
+    const [getRBGeoJson] = useState(regierungsbezirke as FeatureCollection);
+
+    //const [getLKGeoJson] = useState(landkreise as FeatureCollection);
+
+    function GeoContainer({ year }: { year: number }) {
+        const geoJsonRef = useRef<L.GeoJSON>(null);
+        const map = useMap();
+        const groupByAGSFunc = groupBy(GroupBy.AGS);
+        const [getDataRB] = useState<{ [p: string | number]: IDataEntry[] }>(
+            groupByAGSFunc(new DataLoader(GroupBy.AGS).GetDistricts().getDataForYear(year))
+        );
+        const [getDataLK] = useState<{ [p: string | number]: IDataEntry[] }>(
+            groupByAGSFunc(new DataLoader(GroupBy.AGS).GetGovernmentDistricts().getDataForYear(year))
+        );
+
+        const [minRB, maxRB] = useMemo(() => {
+            let minValue = Infinity;
+            let maxValue = -Infinity;
+            Object.values(getDataRB)
+                .flat(1)
+                .forEach((dataEntry) => {
+                    const value = dataEntry.used_area_percent;
+                    minValue = minValue > value ? value : minValue;
+                    maxValue = maxValue < value ? value : maxValue;
+                });
+            return [0, maxValue];
+        }, [getDataRB]);
+
+        const [minLK, maxLK] = useMemo(() => {
+            let minValue = Infinity;
+            let maxValue = -Infinity;
+            Object.values(getDataLK)
+                .flat(1)
+                .forEach((dataEntry) => {
+                    const value = dataEntry.used_area_percent;
+                    minValue = minValue > value ? value : minValue;
+                    maxValue = maxValue < value ? value : maxValue;
+                });
+            return [0, maxValue];
```

Comment:
Same here

Reply:
I didn't know how we calculate the relative distribution, two approaches:
1. MinValue → MaxValue: Results in hard distribution of the individual values  → no homogeneous color gradient
2. 0% → MaxValue: homogeneous color gradient

That's the reason I left the MinValue calculation to switch back if one approach isn't fit our use case.